### PR TITLE
Fixes app crashes reported on Android Version 5

### PR DESCRIPTION
**Pull request by junaidwarsivd**
issue reported on ZEIR https://github.com/opensrp/opensrp-client-path-zeir/issues/185

### CODE REVIEW

File: opensrp-child/src/main/java/org/smartregister/child/model/ChildMotherDetailModel.java [hunk 1]
```diff
@@ -198,7 +207,11 @@ public void setMotherBaseEntityId(String motherBaseEntityId) {
 
     @Override
     public int compareTo(ChildMotherDetailModel childMotherDetailModel) {
-        return this.getZeirId().compareTo(childMotherDetailModel.getZeirId());
+        if(this.getZeirId() != null &&  childMotherDetailModel.getZeirId() != null)
+            return this.getZeirId().compareTo(childMotherDetailModel.getZeirId());
+        else
+            return 0;
```
Comment (ellykits) [inline opensrp-child/src/main/java/org/smartregister/child/model/ChildMotherDetailModel.java:213]: Any reason for the extra null check? Ideally both the identifiers for the child and the mother can never be null. If they are so then the fix should be ensuring the zeir id is never null. Returning 0 means the objects are equal, is that what is desired?

**Review by ellykits — COMMENTED** [review]
What does this implementation mean for existing projects that are not using MVACC? Is it backward compatible?

**Comment by ellykits** [comment]
@junaidwarsivd Rename this PR with a more descriptive name. You can tag the issue it fixes in the description. Also update the version in `gradle.properties` file

File: opensrp-child/src/main/java/org/smartregister/child/model/ChildMotherDetailModel.java (hunk 1 above)
Comment (junaidwarsivd) [reply opensrp-child/src/main/java/org/smartregister/child/model/ChildMotherDetailModel.java:213]: xception: java.lang.NullPointerException: Attempt to invoke virtual method 'int java.lang.String.compareTo(java.lang.String)' on a null object reference
       at org.smartregister.child.model.ChildMotherDetailModel.compareTo(ChildMotherDetailModel.java:201)
       at org.smartregister.child.model.ChildMotherDetailModel.compareTo(ChildMotherDetailModel.java:9)
       at java.util.Collections$ReverseComparator.compare(Collections.java:5165)
       at java.util.Collections$ReverseComparator.compare(Collections.java:5156)
       at java.util.TimSort.countRunAndMakeAscending(TimSort.java:355)
       at java.util.TimSort.sort(TimSort.java:220)
       at java.util.Arrays.sort(Arrays.java:1498)
       at java.util.ArrayList.sort(ArrayList.java:1470)
       at java.util.Collections.sort(Collections.java:201)
       at org.smartregister.child.util.ChildJsonFormUtils.processReturnedAdvanceSearchResults(ChildJsonFormUtils.java:2253)
       at org.smartregister.child.model.BaseChildAdvancedSearchModel.getChildMotherDetailModels(BaseChildAdvancedSearchModel.java:208)
       at org.smartregister.path.model.AdvancedSearchModel.createMatrixCursor(AdvancedSearchModel.java:63)
       at org.smartregister.child.presenter.BaseChildAdvancedSearchPresenter.onResultsFound(BaseChildAdvancedSearchPresenter.java:112)
       at org.smartregister.child.interactor.ChildAdvancedSearchInteractor.lambda$null$0(ChildAdvancedSearchInteractor.java:55)
       at org.smartregister.child.interactor.-$$Lambda$ChildAdvancedSearchInteractor$sKC-cLVEIGA04K-drgD59pcU0cY.run(:6)
       at android.os.Handler.handleCallback(Handler.java:873)
       at android.os.Handler.dispatchMessage(Handler.java:99)
       at android.os.Looper.loop(Looper.java:214)
       at android.app.ActivityThread.main(ActivityThread.java:7156)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:494)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:975)

In firebase this crash was occurring frequently on thats why i've added an extra null check where it was occuring on comparison the goal was if any Odd object with null id occurs then it wont do any sort on the false object @ellykits

**Comment by ellykits** [comment]
Closing this as it is covered here https://github.com/opensrp/opensrp-client-child/pull/268

**Pull request closed without merging**